src/main/java/com/microsoft/azure/functions/worker/reflect/EnhancedClassLoaderProvider.java

```diff
@@ -73,4 +88,5 @@ public void addUrl(URL url) throws IOException {
         urls.add(url);
     }
     private final Set<URL> urls;
+    private static volatile URLClassLoader classLoaderInstance;
```

Comment:
Using the volatile keyword to avoid any cache incoherence issues. Review if this is necessary for our use case